diff -Naurp a/xbmc/cores/VideoPlayer/VideoPlayer.cpp b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2026-06-01 15:29:40.984007695 +0000
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2026-06-01 15:31:05.347574239 +0000
@@ -1535,6 +1535,10 @@ bool CVideoPlayer::ReadPacket(DemuxPacke
       const bool incomingHevcVideo = std::ranges::any_of(
           m_SelectionStreams.Get(StreamType::VIDEO),
           [](const SelectionStream& ss) { return ss.codecId == AV_CODEC_ID_HEVC; });
+
+      if (realtimePlaybackStarted && incomingHevcVideo)
+        m_realtimeExpectVideoStream = true;
+
       const bool forceInternalReset =
           realtimePlaybackStarted && (m_realtimeExpectVideoStream || m_CurrentVideo.id >= 0) &&
           (incomingHevcVideo || m_CurrentVideo.hint.codec == AV_CODEC_ID_HEVC);
@@ -5358,7 +5362,7 @@ bool CVideoPlayer::OpenVideoStream(CDVDS
 
   m_HasVideo = true;
   if (m_pInputStream && m_pInputStream->IsRealtime())
-    m_realtimeExpectVideoStream = true;
+    m_realtimeExpectVideoStream = (hint.codec == AV_CODEC_ID_HEVC);
 
   static_cast<IDVDStreamPlayerVideo*>(player)->SendMessage(
       std::make_shared<CDVDMsg>(CDVDMsg::PLAYER_REQUEST_STATE), 1);
