diff -Naurp kodi/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp kodi-patch/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
--- kodi/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp	2025-06-02 08:34:20.826577909 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp	2025-06-02 08:41:09.000000000 +0000
@@ -200,6 +200,10 @@ void CVideoPlayerAudio::UpdatePlayerInfo
   s << "aq:" << std::setw(2) << std::min(99, m_messageQueue.GetLevel());
   s << "% " << std::fixed << std::setprecision(3) << m_messageQueue.GetTimeSize();
   s << "s, Kb/s:" << std::fixed << std::setprecision(2) << m_audioStats.GetBitrate() / 1024.0;
+  s << ", ac:"   << m_processInfo.GetAudioDecoderName().c_str();
+  if (!m_info.passthrough)
+    s << ", chan:" << m_processInfo.GetAudioChannels().c_str();
+  s << ", " << m_streaminfo.samplerate/1000 << " kHz";
 
   // print a/v discontinuity adjustments counter when audio is not resampled (passthrough mode)
   if (m_synctype == SYNC_DISCON)
diff -Naurp kodi/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp kodi-patch/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp
--- kodi/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp	2025-06-02 06:03:31.000000000 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/VideoPlayerVideo.cpp	2025-06-02 08:44:00.000000000 +0000
@@ -956,11 +956,14 @@ CVideoPlayerVideo::EOutputState CVideoPl
 std::string CVideoPlayerVideo::GetPlayerInfo()
 {
   std::ostringstream s;
+  int width, height;
+  m_processInfo.GetVideoDimensions(width, height);
   s << "vq:" << std::setw(2) << std::min(99, m_processInfo.GetLevelVQ());
   s << "% " << std::fixed << std::setprecision(3) << m_messageQueue.GetTimeSize();
   s << "s, Mb/s:" << std::fixed << std::setprecision(2)
     << static_cast<double>(GetVideoBitrate()) / (1024.0 * 1024.0);
-  s << ", fr:" << std::fixed << std::setprecision(3) << m_fFrameRate;
+  s << ", dc:"   << m_processInfo.GetVideoDecoderName().c_str();
+  s << ", " << width << "x" << height << (m_processInfo.GetVideoInterlaced() ? "i" : "p") << " [" << std::setprecision(2) << m_processInfo.GetVideoDAR() << "]@" << std::fixed << std::setprecision(3) << m_processInfo.GetVideoFps() << ", deint:" << m_processInfo.GetVideoDeintMethod();
   s << ", drop:" << m_iDroppedFrames;
   s << ", skip:" << m_renderManager.GetSkippedFrames();
 
