diff -Naurp inputstream.ffmpegdirect/src/stream/FFmpegStream.cpp inputstream.ffmpegdirect-patch/src/stream/FFmpegStream.cpp
--- inputstream.ffmpegdirect/src/stream/FFmpegStream.cpp	2025-06-21 06:47:53.000000000 +0000
+++ inputstream.ffmpegdirect-patch/src/stream/FFmpegStream.cpp	2025-07-09 12:01:11.000000000 +0000
@@ -892,16 +892,6 @@ bool FFmpegStream::Open(bool fileinfo)
   m_startTime = 0;
   m_seekStream = -1;
 
-  if (m_checkTransportStream && m_streaminfo)
-  {
-    int64_t duration = m_pFormatContext->duration;
-    Dispose();
-    m_reopen = true;
-    if (!Open(false))
-      return false;
-    m_pFormatContext->duration = duration;
-  }
-
   return true;
 }
 
