diff -Naurp ff-base/src/stream/CurlInput.cpp ff-new/src/stream/CurlInput.cpp
--- ff-base/src/stream/CurlInput.cpp	2026-09-30 05:57:28.000000000 +0000
+++ ff-new/src/stream/CurlInput.cpp	2026-10-01 08:49:04.672818814 +0000
@@ -90,6 +90,13 @@ void CurlInput::Close()
   m_eof = true;
 }
 
+void CurlInput::Abort()
+{
+  if (m_pFile)
+    m_pFile->Abort();
+}
+
+
 int CurlInput::Read(uint8_t* buf, int buf_size)
 {
   if (!m_pFile) return -1;
diff -Naurp ff-base/src/stream/CurlInput.h ff-new/src/stream/CurlInput.h
--- ff-base/src/stream/CurlInput.h	2026-09-28 11:04:43.000000000 +0000
+++ ff-new/src/stream/CurlInput.h	2026-10-01 08:48:24.121248287 +0000
@@ -27,6 +27,7 @@ public:
   ~CurlInput();
   bool Open(const std::string& filename, const std::string& mimeType, unsigned int flags);
   void Close();
+  void Abort();
   int Read(uint8_t* buf, int buf_size);
   int64_t Seek(int64_t offset, int whence);
   bool IsEOF();
diff -Naurp ff-base/src/stream/FFmpegStream.cpp ff-new/src/stream/FFmpegStream.cpp
--- ff-base/src/stream/FFmpegStream.cpp	2026-09-30 05:57:28.000000000 +0000
+++ ff-new/src/stream/FFmpegStream.cpp	2026-10-01 08:48:24.121383610 +0000
@@ -270,6 +270,7 @@ void FFmpegStream::DemuxReset()
 
 void FFmpegStream::DemuxAbort()
 {
+  m_curlInput->Abort();
   m_timeout.SetExpired();
 }
 
